pkg/core/resources/apis/hostnamegenerator/api/v1alpha1/hostnamegenerator.go

```diff
@@ -7,7 +7,7 @@ import (
 )
 
 type LabelSelector struct {
-	MatchLabels map[string]string `json:"matchLabels,omitempty"`
+	MatchLabels map[string]string `json:"matchLabels"`
```

Comment:
It was possible to create HG without `MatchLabels`. You need to check if previously created HG can be returned with `matchedLabels: null`. If that's the case then they won't pass OAPI validation